app/core/Snaps/SnapsMethodMiddleware.ts

```diff
@@ -4,16 +4,25 @@ import {
   RequestedPermissions,
   SubjectType,
 } from '@metamask/permission-controller';
+import { SnapRpcHookArgs } from '@metamask/snaps-utils';
 import { RestrictedMethods } from '../Permissions/constants';
+import { keyringSnapPermissionsBuilder } from '../SnapKeyring/keyringSnapsPermissions';
+import { SnapId } from '@metamask/snaps-sdk';
+import { EngineContext } from '../Engine';
+import { handleSnapRequest } from './utils';
 
+export function getSnapIdFromRequest(
+  request: Record<string, unknown>,
+): string | null {
```

Comment:
```suggestion
): SnapId | null {
```

Would remove the need to typecast the result for usage as `SnapId`

Reply:
But we might need to typecast `snapId` here (which might be inferred as `string` while using:

```ts
return typeof snapId === 'string' ? snapId : null;
```

Would need to be changed to:

```ts
return typeof snapId === 'string' ? snapId as SnapId : null;
```

In anyway, I do agree with you, this function should return a `SnapId`, so typecast should happen here rather than elsewhere :)